src/browser/api/window.js

```diff
@@ -1137,26 +1140,39 @@ Window.getParentApplication = function() {
 Window.getParentWindow = function() {};
 
 /**
+ * Sets/updates window's plugin state and emits relevant events
+ */
+Window.setWindowPluginState = function(identity, payload) {
+    const { uuid, name } = identity;
+    const { name: pluginName, version, state, allDone } = payload;
+    const updateTopic = allDone ? 'plugin-state-changed' : 'plugin-state-changing';
+    let { pluginState } = Window.wrap(uuid, name);
+
+    // Single plugin state change
+    if (!allDone) {
+        pluginState = pluginState.find(e => e.name === pluginName && e.version === version);
```

Comment:
what is `e` short for, lets maybe just call it that? (`pluginState` does not sound like an array, not sure what would be in it...)

Reply:
e is for entry?